packages/astro/src/vite-plugin-scanner/scan.ts

```diff
@@ -51,33 +49,41 @@ export async function scan(
 
 	let pageOptions: PageOptions = {};
 	for (const _export of exports) {
-		const { n: name, le: endOfLocalName } = _export;
-		// mark that a `prerender` export was found
-		if (BOOLEAN_EXPORTS.has(name)) {
-			// For a given export, check the value of the local declaration
-			// Basically extract the `const` from the statement `export const prerender = true`
-			const prefix = code
-				.slice(0, endOfLocalName)
-				.split('export')
-				.pop()!
-				.trim()
-				.replace('prerender', '')
-				.trim();
-			// For a given export, check the value of the first non-whitespace token.
-			// Basically extract the `true` from the statement `export const prerender = true`
-			const suffix = code.slice(endOfLocalName).trim().replace(/\=/, '').trim().split(/[;\n]/)[0];
-			if (prefix !== 'const' || !(isTruthy(suffix) || isFalsy(suffix))) {
+		const { n: name, ls: startOfLocalName, le: endOfLocalName } = _export;
+		if (startOfLocalName === -1 || endOfLocalName === -1) continue;
```

Comment:
This only occurs if there is no local variable being exported - i.e. `export default 1`. That's impossible to do in a `.astro` file because it is already exporting the component as default.